src/mesos_state_client.erl

```diff
@@ -36,6 +38,54 @@ maybe_enable_ssl(Options) ->
       Options
   end.
 
+-spec(maybe_use_token(list()) -> list()).
+maybe_use_token(Headers) ->
+  case try_token_from_file() of
+    undefined ->
+      maybe_use_env_token(Headers);
+    {ok, Token} ->
+      [{"Authorization", Token}|Headers]
+  end.
+
+-spec(maybe_use_env_token(list()) -> list()).
+maybe_use_env_token(Headers) ->
```

Comment:
It also might be worth writing a utility function:
`-spec(take_first([fun(() -> term())], default) -> term()`